src/test/renderer.test.cpp

```diff
@@ -66,34 +111,42 @@ void OnButtonClick(void *object, events::ButtonClickEventArgs &event) {
     }
 }
 
-int main(int argc, char *argv[]) {
-
-    Engine engine("Renderer Test", WINDOW_WIDTH, WINDOW_HEIGHT, true);
-
-    // Create texture object.
-    auto texture = engine.MakeGameObject<TextureObject>(
-        "image", Vector3(0, 0, 0), "assets/admirals.png");
-
+void CreateUIElements(Engine &engine,
+                      std::shared_ptr<TextureObject> textureObj) {
     const Vector2 elementSize = Vector2(300, 40);
 
     auto btn1 = engine.MakeUIElement<UI::Button>(
         "btn1", 0, "Move Image Left", elementSize, Color::BLACK, Color::WHITE);
     btn1->onClick.Subscribe(OnButtonClick);
     btn1->onClick.Subscribe(
-        BIND_EVENT_HANDLER_FROM(TextureObject::ButtonClickHandler, texture));
+        BIND_EVENT_HANDLER_FROM(TextureObject::ButtonClickHandler, textureObj));
 
     auto btn2 = engine.MakeUIElement<UI::Button>(
         "btn2", 0, "Move Image Right", elementSize, Color::BLACK, Color::WHITE);
     btn2->onClick.Subscribe(OnButtonClick);
     btn2->onClick.Subscribe(
-        BIND_EVENT_HANDLER_FROM(TextureObject::ButtonClickHandler, texture));
+        BIND_EVENT_HANDLER_FROM(TextureObject::ButtonClickHandler, textureObj));
 
     auto testText1 = engine.MakeUIElement<UI::TextElement>(
-        "text1", 0, "Left aligned", Vector2(220, 40), Color::BLACK);
+        "text1", 0, "Le", Vector2(32, 40), Color::BLACK);
```

Comment:
Was this intentional?

Reply:
Was figuring out values, this can be reversed.